test/goose/brokers/redis/integration_test.clj

```diff
@@ -184,3 +186,84 @@
+                   (c/accumulate-batch-args "arg-one")
+                   (c/accumulate-batch-args "arg-two"))]
+      (reset! perform-batch-fn-executed (promise))
+      (reset! batch-exec-count 0)
+      (reset! batch-args-count (count args))
+      (let [batch-id (:id (c/perform-batch
+                            tu/redis-client-opts
+                            batch-opts
+                            `perform-batch-fn
+                            args))
+            worker (w/start tu/redis-worker-opts)
+            queue (:queue tu/redis-client-opts)]
+        (is (uuid? (UUID/fromString batch-id))
+            "perform-batch should return a map of id with a uuid value")
+        (is (= (deref @perform-batch-fn-executed 100 :e2e-test-timed-out) args)
+            "Worker should execute on the args passed to perform-batch")
+        (is (empty? (redis-cmds/find-in-list
+                      tu/redis-conn
+                      (d/prefix-queue queue)
+                      #(str/includes? % batch-id)
+                      2000))
+            "Ready queue should not contain any jobs that belong to the batch after execution")
+        (w/stop worker)))))
+
+;;; ======= TEST: Batch state creation ==========
+(def batch-id (str (random-uuid)))
+(def batch
+  {:id              batch-id
+   :callback-fn-sym "`prn"
+   :jobs            '({:id             (str (random-uuid))
+                       :execute-fn-sym clojure.core/prn
+                       :args           ["foo"]
+                       :queue          tu/queue
+                       :ready-queue    (d/prefix-queue tu/queue)
+                       :retry-opts     retry/default-opts
+                       :enqueued-at    (u/epoch-time-ms)
+                       :batch-id       batch-id}
+                      {:id             (str (random-uuid))
+                       :execute-fn-sym clojure.core/prn
+                       :args           ["bar"]
+                       :queue          tu/queue
+                       :ready-queue    (d/prefix-queue tu/queue)
+                       :retry-opts     retry/default-opts
+                       :enqueued-at    (u/epoch-time-ms)
+                       :batch-id       batch-id})})
+
+(deftest enqueue-test
+  (testing "Broker enqueues batch atomically a job asynchronously"
+    (let [batch-id (:id batch)
+          batch-state (select-keys batch [:id :callback-fn-sym])
+          batch-state-key (d/prefix-batch batch-id)
+          enqueued-job-set (d/construct-batch-job-set batch-id d/enqueued-job-set)
+          job-ids (map :id (:jobs batch))]
+      (is (= (redis-cmds/exists tu/redis-conn batch-state-key) 0))
+      (is (= (redis-cmds/exists tu/redis-conn enqueued-job-set) 0))
+
+      (redis-batch/enqueue tu/redis-conn batch)
+
+      (is (= (redis-cmds/parse-map tu/redis-conn batch-state-key) batch-state))
+      (is (= (->> enqueued-job-set
+               (redis-cmds/set-members tu/redis-conn)
+               (frequencies))
```

Comment:
question: why do we need frequencies here? the enqueued job collection is a set.

Reply:
As discussed, `frequencies` was being used to compare the flat sequence returned from Redis. 
Since we've updated the implementation of `set-members` to return a set instead, `frequencies` is no longer needed.